packages/cadl-ranch-specs/http/type/model/generic/main.tsp

```diff
@@ -0,0 +1,41 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using Azure.ClientGenerator.Core;
+
+@doc("Illustrates the model generic cases.")
+@scenarioService("/type/model/generic")
+namespace Type.Model.Generic;
+
+model NumericType<T extends numeric> {
+  @doc("This model is used to perform facet processing on fields based on a specified range of values.")
+  values: T[];
+
+  value: T;
+}
+
+@doc("Facets an int32 field by the specified value ranges.")
+model Int32Type extends NumericType<int32> {
+  @doc("The facet type.")
+  kind: "Int32Values";
+}
+
+@scenario
+@scenarioDoc("""
+Expected input body:
+```json
+{
+  "kind": "Int32Values",
+  "values": 
+  [
+    1234
+  ],
+  "value": 1234
+}
+```
+""")
+@route("/genericType")
+@put
+op genericType(@body input: Int32Type): void;
```

Comment:
What is special about those generic types? Is it the template? To note TypeSpec use template(Get fully rendered as different instances) not generic

Reply:
It is the `extends` makes it special. The background is we have some requests [here ](https://github.com/Azure/autorest.csharp/issues/3655)and [here](https://github.com/Azure/autorest.csharp/issues/3263#issuecomment-1786503820). The ideal solution is we could generate a real generic type for `NumericType<T extends numeric>` in C# something like `class NumericType<T> where T is number`. However, architects decided we don't want to generate generic types for now. They decided we only support instantiated type like `Int32Type` in this case. That is why we added `Int32Type` here.

> To note TypeSpec use template(Get fully rendered as different instances) not generic

Will you suggest a different terminology for an instantiated model extending templated model?